pkg/render/render_test.go

```diff
@@ -0,0 +1,292 @@
+			}
+			if tc.imagesContent != "" {
+				file, err := ioutil.TempFile(os.TempDir(), configPath)
+				path := file.Name()
+				assert.NoError(t, err)
+				defer file.Close()
+
+				_, err = file.WriteString(tc.imagesContent)
+				assert.NoError(t, err)
+				r.imagesFile = path
+			}
+
+			if tc.infraContent != "" {
+				file, err := ioutil.TempFile(os.TempDir(), infraPath)
+				path := file.Name()
+				assert.NoError(t, err)
+				defer file.Close()
+
+				_, err = file.WriteString(tc.infraContent)
+				assert.NoError(t, err)
+				r.infrastructureFile = path
+			}
+
+			infra, config, err := r.readAssets()
+			if tc.expectError != "" {
+				assert.EqualError(t, err, tc.expectError)
+			} else {
+				assert.NoError(t, err)
+			}
+			assert.EqualValues(t, config, tc.imagesConfigMap)
+			assert.EqualValues(t, infra, tc.infra)
+		})
+	}
+}
+
+func TestRenderRun(t *testing.T) {
+	tc := []struct {
+		name          string
+		infraContent  string
+		imagesContent string
+		expectObjects []client.Object
+		expectError   string
+	}{{
+		name:          "Unmarshal both infrastructure and images with no issue",
+		infraContent:  infra,
+		imagesContent: imagesConfigMap,
+		expectObjects: aws.GetBootstrapResources(),
+	}, {
+		name:          "Infrastructure not populated",
+		infraContent:  infraMissingPlatform,
+		imagesContent: imagesConfigMap,
+		expectError:   "platform status is not populated on infrastructure",
+	}, {
+		name:          "ImagesConfigMap not located",
+		infraContent:  infra,
+		imagesContent: "BAD",
+		expectError:   "error unmarshaling JSON: while decoding JSON: json: cannot unmarshal string into Go value of type v1.ConfigMap",
+	}}
+
+	infraPath := "infra.yaml"
+	configPath := "imagesConfigMap.yaml"
+
+	for _, tc := range tc {
+		t.Run(tc.name, func(t *testing.T) {
+			imagesFile := "not_found"
+			infrastructureFile := "not_found"
+			destination, err := ioutil.TempDir("", "test")
+			assert.NoError(t, err)
+
+			if tc.imagesContent != "" {
+				file, err := ioutil.TempFile(os.TempDir(), configPath)
+				path := file.Name()
+				assert.NoError(t, err)
+				defer file.Close()
+
+				_, err = file.WriteString(tc.imagesContent)
+				assert.NoError(t, err)
+				imagesFile = path
+			}
+
+			if tc.infraContent != "" {
+				file, err := ioutil.TempFile(os.TempDir(), infraPath)
+				path := file.Name()
+				assert.NoError(t, err)
+				defer file.Close()
+
+				_, err = file.WriteString(tc.infraContent)
+				assert.NoError(t, err)
+				infrastructureFile = path
+			}
+
+			r := New(infrastructureFile, imagesFile)
+			err = r.Run(destination)
+			if tc.expectError != "" {
+				assert.EqualError(t, err, tc.expectError)
+				return
+			}
+			assert.NoError(t, err)
+
+			// Assert all files were written to bootstrap dir
+			files, err := ioutil.ReadDir(path.Join(destination, bootstrapPrefix))
+			assert.NoError(t, err)
+			assert.Len(t, files, len(tc.expectObjects))
+		})
+	}
+}
+
+func TestWriteAssets(t *testing.T) {
```

Comment:
Would be nice to check written file content as well.

Reply:
Done.